pom.xml

```diff
@@ -259,6 +259,59 @@
 
   <build>
     <plugins>
+      <!-- TODO: Remove me when Eclipse recongize generated sources -->
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>add-source</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>add-source</goal>
+            </goals>
+            <configuration>
+              <sources>
+                <source>${project.build.directory}/generated-sources/annotations/</source>
+              </sources>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      
+      <!-- TODO: Remove me when Eclipse recongize generated sources -->
```

Comment:
Are you using the latest Eclipse version?

Reply:
typo "recognize"

Anyway, this sucks. Eclipse could handle maven's generated sources in 2008 so then the created these messy workarounds. It's 2018 now. I can't believe Eclipse hasn't fixed this. I am sure they have - it's probably a setting somewhere that says "do the right thing with maven files" that is defaulted to "false".
Let's not pollute our build with these old workarounds - let's look for today's real solution with eclipse.